tuxemon/technique.py

```diff
@@ -297,25 +297,38 @@ def changestats(self, user: Monster, target: Monster) -> EffectResult:
         '''
         statsmaster = [self.statspeed, self.stathp, self.statarmour, self.statmelee, self.statranged, self.statdodge]
         statslugs = ['speed', 'hp', 'armour', 'melee', 'ranged', 'dodge']
+        newstatvalue = 0
         for stat, slug in zip(statsmaster, statslugs):
             if not stat:
                 continue
-            value = stat['value']
-            dividing = stat['dividing']
+            minvalue = stat['minvalue']
```

Comment:
As the values of the stat are now optional (because `total=False`), you need to use the `get` method with a default value in case they don't exist. If you configure Mypy in your IDE it should flag this as an error.